Resources/doc/configuration_reference.md
CCDNUser MemberBundle Configuration Reference.
==============================================

All available configuration options are listed below with their default values.

``` yml
#
# for CCDNUser MemberBundle    
#
ccdn_user_member:
    template:
        engine:               twig
    seo:
        title_length:         67
    member:
        list:
            layout_template:      CCDNComponentCommonBundle:Layout:layout_body_right.html.twig
            members_per_page:     50
            member_since_datetime_format:  d-m-Y - H:i
            requires_login:       true

```

- [Return back to the docs index](index.md).


Resources/doc/default_fos_configuration.md

``` yml
fos_user:
   db_driver:      orm
   firewall_name:  main
   user_class:     CCDNUser\UserBundle\Entity\User
   use_listener:   true
#    from_email:     { webmaster@example.com: Admin }
   profile:
       form:
#            type:               FOS\UserBundle\Form\ProfileFormType
#            handler:            FOS\UserBundle\Form\ProfileFormHandler
           name:               fos_user_profile_form
           validation_groups:  [Profile]
   change_password:
       form:
#            type:               FOS\UserBundle\Form\ChangePasswordFormType
#            handler:            FOS\UserBundle\Form\ChangePasswordFormHandler
#            name:               fos_user_change_password_form
#            validation_groups:  [ChangePassword]
   registration:
#        confirmation:
#            from_email: ~
#            enabled:    false
#            template:   FOSUserBundle:Registration:email.txt.twig
        form:
            type:               ccdn_user_registration
#            handler:            FOS\UserBundle\Form\RegistrationFormHandler
#            name:               fos_user_registration_form
#            validation_groups:  [Registration]
   resetting:
       token_ttl: 86400
#        email:
#            from_email: ~
#            template:   FOSUserBundle:Resetting:email.txt.twig
#        form:
#            type:               FOS\UserBundle\Form\ResettingFormType
#            handler:            FOS\UserBundle\Form\ResettingFormHandler
#            name:               fos_user_resetting_form
#            validation_groups:  [ResetPassword]
   service:
#        mailer:                 fos_user.util.mailer.default
 #      email_canonicalizer:    fos_user.util.email_canonicalizer.default
 #      username_canonicalizer: fos_user.util.username_canonicalizer.default
       user_manager:           fos_user.user_manager.default
#    encoder:
#        algorithm:          sha512
#        encode_as_base64:   false
#        iterations:         1
   template:
       engine: twig
       theme:  CCDNUserUserBundle:Form:fields.html.twig
#    group:
#        group_class:    ~ # Required when using groups
#        form:
#            type:               FOS\UserBundle\Form\GroupFormType
#            handler:            FOS\UserBundle\Form\GroupHandler
#            name:               fos_user_group_form
#            validation_groups:  [Registration]

```


DependencyInjection/Configuration.php
<?php

/*
 * This file is part of the CCDNUser MemberBundle
 *
 * (c) CCDN (c) CodeConsortium <http://www.codeconsortium.com/>
 *
 * Available on github <http://www.github.com/codeconsortium/>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace CCDNUser\MemberBundle\DependencyInjection;

use Symfony\Component\Config\Definition\ConfigurationInterface;
use Symfony\Component\Config\Definition\Builder\TreeBuilder;
use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;

/**
 * This is the class that validates and merges configuration from your app/config files
 *
 * To learn more see {@link http://symfony.com/doc/current/cookbook/bundles/extension.html#cookbook-bundles-extension-config-class}
 *
 * @author Reece Fowell <reece@codeconsortium.com>
 * @version 1.0
 */
class Configuration implements ConfigurationInterface
{
	
    /**
     * {@inheritDoc}
     */
    public function getConfigTreeBuilder()
    {
        $treeBuilder = new TreeBuilder();
        $rootNode = $treeBuilder->root('ccdn_user_member');

        // Here you should define the parameters that are allowed to
        // configure your bundle. See the documentation linked above for
        // more information on that topic.
        $rootNode
            ->addDefaultsIfNotSet()
            ->canBeUnset()
            ->children()
                ->arrayNode('template')
                    ->addDefaultsIfNotSet()
                    ->canBeUnset()
                    ->children()
                        ->scalarNode('engine')->defaultValue('twig')->end()
                    ->end()
                ->end()
            ->end();

        $this->addSEOSection($rootNode);
        $this->addMemberSection($rootNode);

        return $treeBuilder;
    }

    /**
     *
     * @access protected
     * @param ArrayNodeDefinition $node
     */
    protected function addSEOSection(ArrayNodeDefinition $node)
    {
        $node
            ->addDefaultsIfNotSet()
            ->canBeUnset()
            ->children()
                ->arrayNode('seo')
                    ->addDefaultsIfNotSet()
                    ->canBeUnset()
                    ->children()
                        ->scalarNode('title_length')->defaultValue('67')->end()
                    ->end()
                ->end()
            ->end();
    }

    /**
     *
     * @access private
     * @param ArrayNodeDefinition $node
     */
    private function addMemberSection(ArrayNodeDefinition $node)
    {
        $node
            ->addDefaultsIfNotSet()
            ->canBeUnset()
            ->children()
                ->arrayNode('member')
                    ->addDefaultsIfNotSet()
                    ->canBeUnset()
                    ->children()
                        ->arrayNode('list')
                            ->addDefaultsIfNotSet()
                            ->canBeUnset()
                            ->children()
                                ->scalarNode('layout_template')->defaultValue('CCDNComponentCommonBundle:Layout:layout_body_right.html.twig')->end()
                                ->scalarNode('members_per_page')->defaultValue(50)->end()
                                ->scalarNode('member_since_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                                ->scalarNode('requires_login')->defaultValue(true)->end()
                            ->end()
                        ->end()
                    ->end()
                ->end()
            ->end();
    }
}
